### ISSUE TITLE
Bug: gc shared_log bug

### ISSUE THREAD

**Issue by hycdong** [apache/incubator-pegasus#747]
## Bug Report
In Pegasus version `2.0.0`, we noticed some problems about shared_log garbage collection. In Pegasus, gc shared_log is executed in replication_long thread pool and is expected to be executed every 30 seconds. However, we found that almost 30 minutes gc function is not called, and the result of gc shared_log is not meeting expections. After gc, there are almost 800 files which is greatly exceed the limit count.

In conclusion, gc shared_log may have following problems:
- gc timer may be blocked and not triggered for long time
- gc shared_log has bug if there are too many log files should be gc

**Comment by zhangyifan27** [comment]
This bug could be reproduced in heavy write workload.
I added some logs and found that sometimes `plog->flush_once()` took several seconds:
```
D2021-07-09 16:04:12.656 (1625817852656548085 36002) replica.rep_long0.04010000000001fa: replica_stub.cpp:1723:on_gc(): start to flush private log for replica 282.87@10.132.5.5:32801
D2021-07-09 16:04:27.360 (1625817867360696895 36002) replica.rep_long0.04010000000001fa: replica_stub.cpp:1725:on_gc(): flush private log finish for replica 282.87@10.132.5.5:32801
```
As a result, the gc task may take a long time:
```
D2021-07-09 16:04:30.950 (1625817870950904989 36002) replica.rep_long0.04010000000001fa: replica_stub.cpp:1864:on_gc(): finish to garbage collection, time_used_ns = 18294669709
```
In `mutation_log_private::flush_internal()`, we need to acquire `_plock`, maybe there is lock contention between `mutation_log_private::flush_internal` and 'mutation_log_private::append`.

**Comment by levy5307** [comment]
Shared log was removed before. So close it now.
